bin/monitor.ml

```diff
@@ -57,14 +57,86 @@ module State : sig
   (** Given a state update, render the update. *)
   val render : t -> Update.t -> unit
 end = struct
+  module Job_state = struct
```

Comment:
I don't know how feasible it is, but it would help to keep all the code that I gave you in its own module. It will be easier to sync with upstream that way.

Reply:
Actually, this entire module can live in its own file. I don't see what's the issue with moving it.